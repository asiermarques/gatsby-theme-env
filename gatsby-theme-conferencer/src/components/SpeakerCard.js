import React from "react";

const SpeakerCard = ({ speaker }) => (
  <a className="card speaker" href={speaker.slug}>
    <div
      className="image"
      style={{ backgroundImage: `url('${speaker.image}')` }}
    />
    <div className="card-body">
      <h5 className="card-title">{speaker.name}</h5>
      <p className="card-text">{speaker.company}</p>
    </div>
  </a>
);

export default SpeakerCard;
