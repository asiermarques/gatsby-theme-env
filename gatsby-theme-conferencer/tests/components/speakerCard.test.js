import React from "react";
import { describe, it } from "@jest/globals";
import renderer from "react-test-renderer";

import SpeakerCard from "../../src/components/SpeakerCard";
import SpeakerFixture from "../__fixtures/Speaker";

describe("SpeakerCard", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<SpeakerCard speaker={SpeakerFixture} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
