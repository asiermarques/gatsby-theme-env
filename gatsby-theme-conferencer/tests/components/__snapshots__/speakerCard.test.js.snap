// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakerCard renders correctly 1`] = `
<a
  className="card speaker"
  href="sara-smith"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url('/assets/img/test.jpg')",
      }
    }
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Sara Smith
    </h5>
    <p
      className="card-text"
    >
      CTO Tyller Corp.
    </p>
  </div>
</a>
`;
